test/runtime/memcache.go

```diff
@@ -106,99 +109,200 @@ var _ = Describe("RuntimeMemcacheBinary", func() {
+	setKeyText := func(key, value string) *helpers.CmdRes {
+		logger.Infof("Setting %s as %s", key, value)
+		cmd := fmt.Sprintf(
+			`sh -c 'echo -en "set %s 0 500 %d\r\n%s\r\nquit\r\n" | nc %s 11211'`, key, len(value), value, memcacheIP)
+		res := vm.ContainerExec("memcache-text-client", cmd)
+		return res
+	}
 
-		By("Getting multiple keys")
-		code := "print(client.get_multi([\"allowed\", \"disallowed\"]))"
-		r = runWithClient(code)
+	getKeyText := func(key string) *helpers.CmdRes {
+		logger.Infof("Getting %s", key)
+		res := vm.ContainerExec("memcache-text-client", fmt.Sprintf(
+			`sh -c 'echo -en "get %s\r\nquit\r\n" | nc %s 11211'`, key, memcacheIP))
+		return res
+	}
 
-		r.ExpectFail("Unable to get multiple keys")
+	Context("Testing text memcache", func() {
+		It("Tests basic memcache operation", func() {
+			key := "test1"
+			value := "value1"
+			By("Setting memcache value")
+			r := setKeyText(key, value)
+			r.ExpectSuccess("Unable to set key %s", key)
+			r.ExpectContains("STORED", "value not stored")
+
+			By("getting memcache value")
+			r = getKeyText(key)
+			r.ExpectSuccess("Unable to get key %s", key)
+			r.ExpectContains(value, "Value mismatch on get")
+			r.ExpectContains("VALUE", "Did not have text VALUE header")
+		})
+
+		It("Tests policy allowing all actions", func() {
+			_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-allow.json"), 300)
+			Expect(err).Should(BeNil(), "Failed to import policy")
+
+			key := "test2"
+			value := "value2"
+			By("Setting memcache value")
+			r := setKeyText(key, value)
+			r.ExpectSuccess("Unable to set key, %s", key)
+			r.ExpectContains("STORED", "value not stored")
+
+			By("getting memcache value")
+			r = getKeyText(key)
+			r.ExpectSuccess("Unable to get key, %s", key)
+			r.ExpectContains(value)
+			r.ExpectContains("VALUE", "Did not have text VALUE header")
+		})
+
+		It("Tests policy disallowing set action", func() {
+			By("Setting key before disallowing set")
+			keyBeforePolicy := "before"
+			valBeforePolicy := "beforeval"
+			r := setKeyText(keyBeforePolicy, valBeforePolicy)
+			r.ExpectSuccess("Unable to set key %q without policy loaded", keyBeforePolicy)
+			r.ExpectContains("STORED", "value not stored")
+
+			By("Importing policy disallowing set")
+			_, err := vm.PolicyImportAndWait(vm.GetFullPath("Policies-memcache-disallow-set.json"), 300)
+			Expect(err).Should(BeNil(), "Failed to import policy")
+
+			By("Trying to set new key")
+			key := "keyAfterPolicy"
+			value := "valAfterPolicy"
+
+			By("Setting memcache value")
+			r = setKeyText(key, value)
+			r.ExpectSuccess()
+			r.ExpectContains("access denied", "did not get access denied message")
+
+			By("getting memcache value")
+			r = getKeyText(keyBeforePolicy)
+			r.ExpectSuccess("Unable to get key set before applying policy")
+			r.ExpectContains(valBeforePolicy, "Value mismatch on get")
+		})
+
+		It("Tests policy allowing actions only for key", func() {
```

Comment:
Clarify this by making it "Tests policy allowing actions for allow key" or something similar - specifying which key makes it clearer.